lib/msf/core/exploit/remote/http/http_cookie.rb

```diff
@@ -253,8 +253,12 @@ def valid_for_uri?(uri)
           end
 
           # Tests if it is OK to accept this cookie if it is sent from the passed +uri+.
+          #
+          # @param [String] uri The uri that will be checked
+          # @return [Boolean] True if the URI is an acceptable URI, false if the URI is nil or resolves to a blank string
           def acceptable_from_uri?(uri)
             return false if uri.nil?
+            return false if URI(uri.strip).host == ''
```

Comment:
Cintext: It looks like the parsing logic of URI has changed in Ruby 3.2

In ruby less than 3.2 the host is nil

```ruby
>> URI('http://').host
=> nil
```

And in ruby 3.2 the host is an empty string
```
3.2.0 :009 > URI('http://').host
""
```

Which broke the following test:

```ruby
it 'will return false if url with http protocol is passed but has no host' do
  protcol = 'http://'

  v = cookie.acceptable_from_uri?(protcol)

  expect(URI(protcol).is_a?(::URI::HTTP)).to eq(true)
  expect(v).to eq(false)
end
```

The test itself is questionable, but the implementation has been updated regardless :+1: